tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala

```diff
@@ -167,6 +167,76 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
     addAll(zScoreMergeJobs.values)
     addAll(windowStouffJobs.values.flatMap(_.values))
     addAll(thresholdJobs.values.flatMap(_.values))
+
+    val xhmmZGzip = xhmmZJobs map {
+      case (sample, jobAndRefFile) =>
+        val sort = sortBgzipAndTabix(jobAndRefFile._2)
+        val pipeJob = new BiopetFifoPipe(this, sort.sortJob :: sort.bgzipJob :: Nil)
+        add(sort.tabixJob)
+        add(pipeJob)
+        sample -> sort
+    }
+
+    val wisecondorZGzip = wisecondorZJobs map {
+      case (sample, job) =>
+        val sort = sortBgzipAndTabix(job.output)
+        val pipeJob = new BiopetFifoPipe(this, sort.sortJob :: sort.bgzipJob :: Nil)
+        add(pipeJob)
+        add(sort.tabixJob)
+        sample -> sort
+    }
+
+    val stouffGzip = windowStouffJobs map {
+      case (sample, subMap) =>
+        val nMap = subMap map {
+          case (window, stouffJob) =>
+            val sort = sortBgzipAndTabix(stouffJob.output)
+            val pipeJob = new BiopetFifoPipe(this, sort.sortJob :: sort.bgzipJob :: Nil)
+            add(pipeJob)
+            add(sort.tabixJob)
+            window -> sort
+        }
+        sample -> nMap
+    }
+
+    thresholdJobs foreach {
+      case (sample, subMap) =>
+        subMap foreach {
+          case (window, threshJob) =>
+            val imageDir = new File(threshJob.output.get.getParentFile, "plots")
+            val sort = sortBgzipAndTabix(threshJob.output.get)
+            val pipeJob = new BiopetFifoPipe(this, sort.sortJob :: sort.bgzipJob :: Nil)
+            add(pipeJob)
+            add(sort.tabixJob)
+            val xhmmZ = xhmmZGzip(sample)
+            val wiseZ = wisecondorZGzip(sample)
+            val stouff = stouffGzip(sample)(window)
+            val plotter = new TarmacPlot(this)
+            plotter.outputDir = imageDir
+            plotter.callFile = sort.bgzipJob.output
+            plotter.stouffFile = stouff.bgzipJob.output
+            plotter.xhmmFile = xhmmZ.bgzipJob.output
+            plotter.wisecondorFile = wiseZ.bgzipJob.output
+            plotter.deps ++= sort.tabixJob.outputIndex :: xhmmZ.tabixJob.outputIndex :: wiseZ.tabixJob.outputIndex :: stouff.tabixJob.outputIndex :: Nil
+            add(plotter)
+        }
+    }
+  }
+
+  case class SortBgzipTabix(sortJob: BedtoolsSort, bgzipJob: Bgzip, tabixJob: Tabix)
+
+  def sortBgzipAndTabix(file: File): SortBgzipTabix = {
+    val sorter = new BedtoolsSort(this)
+    sorter.input = file
+    sorter.isIntermediate = true
+    val sortFile = swapExt(file.getParentFile, file, ".bed", ".sorted.bed")
+    sorter.output = sortFile
+    val gzFile = swapExt(file.getParentFile, file, ".bed", ".sorted.bed.gz")
+    val bgzip = new Bgzip(this)
```

Comment:
This is done in all cases this function is called. The reason not do this in this function is that `pipeJobs` in a `BiopetFifoPipe` is only populated _after_ its `beforeGraph` is called, and I want to access its constituent jobs before that time.